libkbfs/disk_journal.go

```diff
@@ -98,7 +100,22 @@ func (j diskJournal) readOrdinal(path string) (journalOrdinal, error) {
 
 func (j diskJournal) writeOrdinal(
 	path string, o journalOrdinal) error {
-	return ioutil.WriteFile(path, []byte(o.String()), 0600)
+	// Don't use ioutil.WriteFile because it truncates the file first,
+	// and if there's a crash it will leave the journal in an unknown
+	// state.
```

Comment:
Seems very unlikely given the size of the ordinals and the usual size of OS writes, but I guess it's technically possible.  Also, this function is doing nearly the exactly the same thing as `ioutil.WriteFile`, so there's no new risk here.